### PR TITLE
feature/transcript-item

### PR DESCRIPTION
**Pull request by tohuynh**
**Pull request recommendations:**
- [x] Name your pull request _your-development-type/short-description_. Ex: _feature/read-tiff-files_
- [x] Link to any relevant issue in the PR description. Ex: _Resolves [gh-12], adds tiff file format support_
  WIP #2 
- [x] Provide context of changes.
  Working toward the Search transcript component. This PR has a component for a single transcript item. It can be used in the Search transcript component and the Full transcript component.
  Notes
  Installed [react-highlight-words](https://www.npmjs.com/package/react-highlight-words) for search query highlighting
  Added a constant for some media breakpoints (Used them in HomeSearchBar as well)
- [ ] Provide relevant tests for your feature or bug fix.
- [x] Provide or update documentation for any feature added by your pull request.
  Stories [preview](https://tohuynh.github.io/cdp-frontend/?path=/story/library-details-transcript-item--with-picture-src)
  Transcript item with speaker picture src
  Transcript item  with unknown speaker and default avatar (Used heroicons' `user-circle`)
  Transcript item with search query highlights
  Transcript item without `Transcript` click. (This will be used in the full transcript component).
  
Thanks for contributing!

### CODE REVIEW

File: src/styles/mediaBreakpoints.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// px values of different screen widths
+export const screenWidths = {
+  mobile: "320px",
+  tablet: "768px",
+  desktop: "1224px",
+};
```
Comment (JacksonMaxfield) [inline src/styles/mediaBreakpoints.ts:6]: Nice. Appreciate centralizing these.

**Review by JacksonMaxfield — COMMENTED** [review]
Couple of comments:

I am wondering if we can add icons next to the video clip and the transcript text OR have them wrapped in some sort of button "look". Or maybe have text that says "Jump to section: " and then those buttons. Currently it isn't immediately obvious what those buttons _do_ or that they are buttons and not just highlighted text?

Any other options for the "Unknown Speaker" icon? Not my favourite. I can look around too.

And I guess it's fine to keep the functionality but I don't think we would ever be missing a timestamped transcript.

File: package.json [hunk 2]
```diff
@@ -115,6 +116,7 @@
   "dependencies": {
```
[the remark this answers is not in the document]
Comment (JacksonMaxfield) [reply package.json:116]: CI / GitHub generally picks up when vulns or breaking changes occur.

**Comment by tohuynh** [comment]
>I am wondering if we can add icons next to the video clip and the transcript text OR have them wrapped in some sort of button "look". Or maybe have text that says "Jump to section: " and then those buttons. Currently it isn't immediately obvious what those buttons do or that they are buttons and not just highlighted text?

I like the play icon idea.

>Any other options for the "Unknown Speaker" icon? Not my favourite. I can look around too.

You are not a fan of the `user` icon? I can't think of anything else to represent a generic avatar.
The other option is to use user without the circle.
image

File: src/components/Details/TranscriptItem/TranscriptItem.tsx [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  objectPosition: "center",
+  borderRadius: "50%",
+  width: AVATAR_SIZE,
+  height: AVATAR_SIZE,
+});
+const DefaultAvatarContainer = styled.div({
+  width: AVATAR_SIZE,
+  height: AVATAR_SIZE,
+});
+const DEFAULT_AVATAR = (
```
[the remark this answers is not in the document]
Comment (tohuynh) [reply src/components/Details/TranscriptItem/TranscriptItem.tsx:59]: Yes

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (tohuynh) [reply package.json:116]: I think we are OK with using caret (`^`) since it won't introduce any major breaking changes.

**Comment by tohuynh** [comment]
> that they are buttons and not just highlighted text?

Currently, they are clickable `div`s. Should we change them to `button` and use Mozilla protocol styling?

**Comment by JacksonMaxfield** [comment]
> You are not a fan of the user icon, in general? I can't think of anything else to represent a generic avatar.
> The other option is to use user without the circle.

Hmmm maybes its just that the icon line width is really big. It just seems _heavy_.

> Currently, they are clickable divs. Should we change them to button and use Mozilla protocol styling?

Yea probably. I think the `button` tag for html is also better for accessibility.

**Comment by tohuynh** [comment]
image
Does this look any better? There are fewer lines, but still the same line width (I'm just copying the SVG from hero icons).

**Comment by JacksonMaxfield** [comment]
>  Does this look any better? There are fewer lines, but still the same line width (I'm just copying the SVG from hero icons).

Yea I think so.

**Pull request merged**